src/queue/priority.rs

```diff
@@ -70,11 +70,39 @@ where
     }
 }
 
+/// Priority represents the priority of a task.
+/// Tasks with smaller priority value will be scheduled earlier than bigger ones.
+#[derive(PartialEq, Copy, Clone, Eq, PartialOrd)]
+pub struct Priority {
+    /// The priority of the group this task belongs to, ordered in descending order.
+    pub group_priority: u64,
+    /// The priority value of this task.
+    pub value: u64,
+}
+
+impl Default for Priority {
+    fn default() -> Self {
+        Priority {
+            group_priority: 8,
+            value: 0,
+        }
+    }
+}
+
+impl Ord for Priority {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
```

Comment:
What's the difference from `#[derive(Ord)]`?

Reply:
note that `group_prioirty` is in descending order